src/settings/settings.ts

```diff
@@ -238,6 +238,12 @@ export function getXMLSettings(javaHome: string | undefined, logfile: string, ex
       xml['xml']['catalogs'].push(catalog);
     }
   })
+  externalXmlSettings.xmlFileAssociations.forEach(element => {
```

Comment:
actually wait, this should be handled under the variable substitution code in `variableSubstitution.ts`. Let me take a look at what's happening...

Reply:
Sorry, I misunderstood what this PR addressed. Taking a look into testing this now...